### PR TITLE
Wave and rocket reactions

### PR DESCRIPTION
**Pull request by Lunarmagpie**
Wave when people join the server, and the amazing rust rocket

### CODE REVIEW

**Review by CircuitSacul — APPROVED** [review]

**Review by Lunarmagpie — COMMENTED** [review]
I think this is ready to merge

File: mcodingbot/plugins/reactions.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import hikari
+
+RUST_REGEX = re.compile(r"\brust\b", flags=re.I)
+
+plugin = crescent.Plugin("reactions")
+
+
+@plugin.include
+@crescent.event
+async def on_message(event: hikari.MessageCreateEvent) -> None:
```
Comment (CircuitSacul) [inline mcodingbot/plugins/reactions.py:13]: hikari.GuildMessageCreateEvent?

File: mcodingbot/plugins/reactions.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    message = event.message
+
+    if message.type == hikari.MessageType.GUILD_MEMBER_JOIN:
+        await message.add_reaction("👋")
+        return
+
+    if not message.content:
+        return
+
+    if RUST_REGEX.search(message.content) and "🚀" in message.content:
```
Comment (CircuitSacul) [inline mcodingbot/plugins/reactions.py:23]: Is regex actually faster? Why not "rust" in message.content?

**Review by CircuitSacul — CHANGES_REQUESTED** [review]

File: mcodingbot/plugins/reactions.py (hunk 2 above)
Comment (Lunarmagpie) [reply mcodingbot/plugins/reactions.py:23]: i need `\b` from regex to make it feel right

File: mcodingbot/plugins/reactions.py (hunk 1 above)
Comment (Lunarmagpie) [reply mcodingbot/plugins/reactions.py:13]: Any reason for it to not work in DMs? Other commands do.

Comment (CircuitSacul) [reply mcodingbot/plugins/reactions.py:13]: This isn't a command though? I guess it's ok for Rust :rocket: to work in DMs.

I'm fine either way.

File: mcodingbot/plugins/reactions.py (hunk 2 above)
Comment (CircuitSacul) [reply mcodingbot/plugins/reactions.py:23]: Gotcha

File: mcodingbot/plugins/reactions.py (hunk 1 above)
Comment (Lunarmagpie) [reply mcodingbot/plugins/reactions.py:13]: the ping command works in DMs

Comment (CircuitSacul) [reply mcodingbot/plugins/reactions.py:13]: Are you sure? Pretty sure discord added `dm_permissions` which is False by default.

Comment (CircuitSacul) [reply mcodingbot/plugins/reactions.py:13]: Nope you're right, nevermind lol.

**Review by CircuitSacul — APPROVED** [review]

**Pull request merged**